bliss/surveys/dc2.py

```diff
@@ -293,6 +186,160 @@ def test_dataloader(self):
+            psf_params,
+        ) = load_image_and_catalog(
+            image_idx,
+            self.image_files,
+            self.bg_files,
+            self.image_lim,
+            self.bands,
+            self.n_bands,
+            self.cat_path,
+            self.min_flux_threshold,
+        )
+        return {
+            "tile_catalog": tile_dict,
+            "image": torch.from_numpy(image),
+            "background": torch.from_numpy(bg),
+            "match_id": match_id,
+            "full_catalog": full_cat,
+            "wcs": wcs,
+            "psf_params": psf_params,
+        }
+
+
+def squeeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "1 h w nh nw -> h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "1 h w -> h w")
+    return tile_dict_copy
+
+
+def unsqueeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "h w nh nw -> 1 h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "h w -> 1 h w")
+    return tile_dict_copy
+
+
+def load_image_and_catalog(
+    image_idx, image_files, bg_files, image_lim, bands, n_bands, cat_path, min_flux_threshold
+):
+    image_list, bg_list, wcs_header_str = read_frame_for_band(
+        image_files, bg_files, image_idx, n_bands, image_lim
+    )
+    wcs = from_wcs_header_str_to_wcs(wcs_header_str)
+    image = np.stack(image_list)
+    bg = np.stack(bg_list)
+
+    plocs_lim = image[0].shape
+    height = plocs_lim[0]
+    width = plocs_lim[1]
+    full_cat, psf_params, match_id = Dc2FullCatalog.from_file(
+        cat_path, wcs, height, width, bands, min_flux_threshold
+    )
+    tile_dict = full_cat.to_tile_catalog(4, 5).get_brightest_sources_per_tile().to_dict()
+    tile_dict = squeeze_tile_dict(tile_dict)
+    tile_dict["star_fluxes"] = tile_dict["star_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_fluxes"] = tile_dict["galaxy_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 3] = tile_dict["galaxy_params"][..., 3].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 5] = tile_dict["galaxy_params"][..., 5].clamp(min=1e-18)
+
+    return image, bg, tile_dict, full_cat, wcs, wcs_header_str, match_id, psf_params  # noqa: WPS227
+
+
+def generate_split_file(image_index, self_copy):
+    split_count = 0
+    (
+        image,
+        bg,
+        tile_dict,
+        _,
+        _,
+        wcs_header_str,
+        _,
+        psf_params,
+    ) = load_image_and_catalog(
+        image_index,
+        self_copy["image_files"],
+        self_copy["bg_files"],
+        self_copy["image_lim"],
+        self_copy["bands"],
+        self_copy["n_bands"],
+        self_copy["cat_path"],
+        self_copy["min_flux_threshold"],
+    )
```

Comment:
So many arguments/return values/lines. Can this be refactored somehow?

Reply:
return a dict to address this